### PR TITLE
Visual for *sigh plus a bonus *glasses emote

### PR DESCRIPTION
**Pull request by desertmote**
## About The Pull Request

Some visual effects I've had for over a year, has existed for an equal amount of time on a downstream.
So I finally decided to bring them here, I have no expectations for it though. I just wanted to share it, lets see where it goes.

## Why It's Good For The Game

https://user-images.githubusercontent.com/77534246/190883390-5b61edfa-1b9d-42a4-a51a-18a8b7e0f12f.mp4

(Only *sigh and *glasses are in this PR, the rest should be made into more fitting styles at later date)

## Changelog

:cl:
add: Adds an emote to push up glasses with *glasses
add: Adds a visual effect to *sigh
/:cl:

### CODE REVIEW

**Comment by TheBoondock** [comment]
These looks so good 👍

**Comment by Mothblocks** [comment]
i don't think they should loop 3 times

File: code/modules/mob/living/emote.dm [hunk 1]
```diff
@@ -32,6 +32,112 @@
...
+
+/datum/emote/living/carbon/human/glasses
+	key = "glasses"
+	key_third_person = "glasses"
+	message = "pushes up their glasses."
+	cooldown = 2 SECONDS
+
+/datum/emote/living/carbon/human/glasses/can_run_emote(mob/user, status_check = TRUE, intentional)
+	var/obj/O = user.get_item_by_slot(ITEM_SLOT_EYES)
+	if(!(istype(O, /obj/item/clothing/glasses/regular)))
```
Comment (Mothblocks) [inline code/modules/mob/living/emote.dm:128]:
```suggestion
	var/obj/eyes_slot = user.get_item_by_slot(ITEM_SLOT_EYES)
	if(!(istype(eyes_slot, /obj/item/clothing/glasses/regular)))
```

**Comment by Wallemations** [comment]
I feel like the exclaim and question ones are really big but other than that these look pretty nice
I can't say I understand the annoyed one though. What's that blob supposed to represent? When I personally think annoyed I think a symbol like this

image

**Comment by Zirok-BYOND** [comment]
I have no idea what most of these are supposed to be without looking at the emote text.

**Comment by Troxonater** [comment]
Literally made signed into this account and made my first comment to let you know that the annoyed one is terrible and anyone seeing it for the first time would be very confused. I would take the recommendation of @Wallemations , every other one is good imo.

**Comment by Twaticus** [comment]
i'd rather visual effects for emotes be something the character can actually express, like the current blush or cry. so sigh and glasses seem like the only feasible ones here. things like sweat drops, exclamation, or question marks should be kept to the speech bubbles format.

**Comment by theOOZ** [comment]
> i'd rather visual effects for emotes be something the character can actually express, like the current blush or cry. so sigh and glasses seem like the only feasible ones here. things like sweat drops, exclamation, or question marks should be kept to the speech bubbles format.

👍 will make edits. id like to argue in favor for the sweatdrop though, its a classic. works well even without the sfx

**Comment by theOOZ** [comment]
> i don't think they should loop 3 times

I can't, its not a stylistic choice. `flick_overlay_static` dials in half-way through the iconstate. If I do not loop them, you would only see its final frame. (It would not even be animated)
This is something I messed around with for a loooong time when initially developed.

**Comment by Mothblocks** [comment]
Can you try adding more, empty frames at the end? It looks pretty bad with the loop

**Comment by Farquaar** [comment]
Annoyed kind of reminds me of Calvin and Hobbes, which I like
image

**Comment by Mothblocks** [comment]
> Can you try adding more, empty frames at the end? It looks pretty bad with the loop

Did you try this?

**Comment by theOOZ** [comment]
> > Can you try adding more, empty frames at the end? It looks pretty bad with the loop
> 
> Did you try this?

not yet, will make an attempt very soon.

**Comment by Mothblocks** [comment]
Been stale and conflicting for a while, feel free to reopen when you finish

**Pull request closed without merging**